pom.xml

```diff
@@ -82,16 +135,89 @@
                 </executions>
             </plugin>
             <plugin>
-                <groupId>org.eluder.coveralls</groupId>
-                <artifactId>coveralls-maven-plugin</artifactId>
-                <version>4.3.0</version>
-                <configuration>
-                    <!-- TODO: Don't commit repoToken pubically! Find a way to integrate securely -->
-                    <repoToken>yourcoverallsprojectrepositorytoken</repoToken>
-                    <jacocoReports>
-                        <param>${project.reporting.outputDirectory}/jacoco-ut/jacoco.xml</param>
-                    </jacocoReports>
-                </configuration>
+              <groupId>org.eluder.coveralls</groupId>
+              <artifactId>coveralls-maven-plugin</artifactId>
+              <version>4.3.0</version>
+              <configuration>
+                <!-- TODO: Don't commit repoToken pubically! Find a way to integrate securely -->
+                <repoToken>yourcoverallsprojectrepositorytoken</repoToken>
+                <jacocoReports>
+                  <param>${project.reporting.outputDirectory}/jacoco-ut/jacoco.xml</param>
+                </jacocoReports>
+              </configuration>
+            </plugin>
+            <plugin>
+              <groupId>org.apache.maven.plugins</groupId>
+              <artifactId>maven-source-plugin</artifactId>
+              <version>2.2.1</version>
+              <executions>
+                <execution>
+                  <id>attach-sources</id>
+                  <goals>
+                    <goal>jar-no-fork</goal>
+                  </goals>
+                </execution>
+              </executions>
+            </plugin>
+            <!-- <plugin> -->
```

Comment:
We will enable java-doc plugin when the generator is fully documenting resources and client.